### PR TITLE
Migrate to built-in Neo4j schema

### PR DESCRIPTION
**Pull request by pstutz**
Removes our own schema procedure and uses the built-in one from Neo4j instead.

### CODE REVIEW

File: okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala [hunk 1]
```diff
@@ -27,91 +27,185 @@
...
 
-import scala.collection.JavaConverters._
 import scala.util.{Failure, Success, Try}
 
 object SchemaFromProcedure extends Logging {
 
-  val schemaProcedureName = "org.opencypher.okapi.procedures.schema"
+  /**
+    * Returns the schema for a Neo4j graph. When the schema contains a property that is incompatible with CAPS,
+    * then an exception is thrown. Please set `omitImportFailures` in order to omit such properties from the schema
```
Comment (s1ck) [inline okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala]:
```suggestion
    * an exception is thrown. Please set `omitImportFailures` in order to omit such properties from the schema
```

File: okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala [hunk 2]
```diff
@@ -27,91 +27,185 @@
...
+    * instead.
+    *
+    * This method relies on the built-in Neo4j schema procedures, which were introduced in Neo4j versions 3.3.9,
```
Comment (s1ck) [inline okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala]: We could name those procedures or even link to the Neo4j docs

File: okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala [hunk 3]
```diff
@@ -27,91 +27,185 @@
...
+          val nodeRowsGroupedByCombo = nodeRows.groupBy(_.labels)
+          val nodeSchema = nodeRowsGroupedByCombo.foldLeft(Schema.empty) { case (currentSchema, (combo, rows)) =>
+            currentSchema.withNodePropertyKeys(combo, propertyKeysForRows(rows))
+          }
+          val relRows = cypher(s"CALL $relSchemaProcedure")
+          val relRowsGroupedByType = relRows.groupBy(_.relType)
+          val relSchema = relRowsGroupedByType.foldLeft(Schema.empty) { case (currentSchema, (tpe, rows)) =>
+            currentSchema.withRelationshipPropertyKeys(tpe, propertyKeysForRows(rows))
+          }
+          nodeSchema ++ relSchema
```
Comment (s1ck) [inline okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala]: so nice :)

File: okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala [hunk 4]
```diff
@@ -27,91 +27,185 @@
...
+
+    def isMandatory: Boolean = row.get("mandatory").collect { case CypherBoolean(b) => b }.getOrElse(false)
+
+    /**
+      * Returns the Cypher type from a Neo4j schema row.
+      *
+      * CAPS can have at most one property type for a given property on a given label combination.
+      * If different property types are present on the same property with multiple nodes that have the same
+      * label combination, then by default a schema exception is thrown. If `omitImportFailures` is set, then the
+      * problematic property is instead excluded from the schema and a warning is logged.
```
Comment (s1ck) [inline okapi-neo4j-io/src/main/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedure.scala]: :+1:

File: okapi-neo4j-io-testing/src/test/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedureTest.scala [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+import org.neo4j.graphdb.GraphDatabaseService
+import org.neo4j.harness.{EnterpriseTestServerBuilders, ServerControls}
+import org.opencypher.okapi.api.schema.Schema
+import org.opencypher.okapi.api.types._
+import org.opencypher.okapi.impl.exception.SchemaException
+import org.opencypher.okapi.neo4j.io.Neo4jHelpers._
+import org.opencypher.okapi.testing.BaseTestSuite
+import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll}
+
+class SchemaFromProcedureTest extends BaseTestSuite with BeforeAndAfter with BeforeAndAfterAll {
```
[the remark this answers is not in the document]
Comment (pstutz) [reply okapi-neo4j-io-testing/src/test/scala/org/opencypher/okapi/neo4j/io/SchemaFromProcedureTest.scala]: I moved some over from the previous PR [here](https://github.com/opencypher/cypher-for-apache-spark/pull/753/commits/51c405585a0260826e36a9242f33f51d4948303f).

I didn't find any tests for `SchemaFromProcedure` itself, please let me know if I missed them.

**Comment by pstutz** [comment]
Because CI is down I ran the tests locally before the merge.

**Pull request merged**